src/bundle/Controller/ContentController.php

```diff
@@ -381,4 +383,72 @@ public function updateMainTranslationAction(Request $request): Response
 
         return $this->redirectToRoute('ezplatform.dashboard');
     }
+
+    /**
+     * @param Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function updateVisibilityAction(Request $request): Response
+    {
+        $form = $this->createForm(ContentVisibilityUpdateType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $this->submitHandler->handle($form, function (ContentVisibilityUpdateData $data) {
+                $contentInfo = $data->getContentInfo();
+                $newValue = $data->getVisible();
+
+                if ($contentInfo->isHidden && $newValue === false) {
+                    return $this->notificationHandler->success(
+                        $this->translator->trans(
+                            /** @Desc("Content '%name%' was already hidden.") */
+                            'content.hide.already_hidden',
+                            ['%name%' => $contentInfo->name],
+                            'content'
+                        )
+                    );
+                }
+
+                if (!$contentInfo->isHidden && $newValue === true) {
+                    return $this->notificationHandler->success(
+                        $this->translator->trans(
+                            /** @Desc("Content '%name%' was already visible.") */
+                            'content.reveal.already_visible',
+                            ['%name%' => $contentInfo->name],
+                            'content'
+                        )
+                    );
+                }
+
+                if ($newValue === false) {
+                    $this->contentService->hideContent($contentInfo);
+
+                    return $this->notificationHandler->success(
+                        $this->translator->trans(
+                            /** @Desc("Content '%name%' has been hidden.") */
+                            'content.hide.success',
+                            ['%name%' => $contentInfo->name],
+                            'content'
+                        )
+                    );
+                }
+
+                if ($newValue === true) {
+                    $this->contentService->revealContent($contentInfo);
+
+                    return $this->notificationHandler->success(
+                        $this->translator->trans(
+                            /** @Desc("Content '%name%' has been revealed.") */
+                            'content.reveal.success',
+                            ['%name%' => $contentInfo->name],
+                            'content'
+                        )
+                    );
+                }
+            });
+        }
+
+        return $this->redirectBack($request);
```

Comment:
See my comment below. You can use https://github.com/ezsystems/ezplatform-admin-ui/blob/master/src/bundle/Controller/Location/TrashLocationWithAssetController.php#L67 as a reference.